concrete/controllers/backend/messenger.php

```diff
@@ -75,7 +76,7 @@ public function consume()
             $worker->run();
             return $this->responseFactory->createResponse();
         }
-        throw new \Exception(t('Access Denied'));
+        return new JsonResponse(t('Access Denied'), 401);
```

Comment:
We can also throw a UserMessageException (which should render as a json response in case of ajax requests)

Reply:
Yeah but it shows the exception page for non-ajax requests, so I prefer the current way.